File: bench/ports/kotlin/lists.kt
fun main() {
    val xs = ArrayList<Long>()
    for (i in 0 until 1000000L) { xs.add(i % 1000) }
    val doubled = xs.map { it * 2 }
    val big = doubled.filter { it > 1000 }
    println(big.fold(0L) { acc, n -> acc + n })
}

File: bench/ports/rust/lists.rs
fn main() {
    let mut xs: Vec<i64> = Vec::new();
    for i in 0..1_000_000i64 { xs.push(i % 1000); }
    let doubled: Vec<i64> = xs.iter().map(|it| it * 2).collect();
    let big: Vec<i64> = doubled.iter().filter(|&&it| it > 1000).cloned().collect();
    let acc: i64 = big.iter().fold(0i64, |acc, n| acc + n);
    println!("{}", acc);
}
